test/stress/generate-report/download_results.py

```
@@ -0,0 +1,103 @@
+        path = url[idx + len('gcsweb.k8s.io/gcs/'):]
+        return f"https://storage.googleapis.com/{path}"
+        
+    if url.startswith('https://storage.googleapis.com/'):
+        return url
+        
+    raise ValueError(f"Unsupported URL format: {url}")
+
+def download_file(url, out_path):
+    print(f"Downloading {url} ...")
+    try:
+        with urllib.request.urlopen(url) as response:
```

Ruff (0.15.21)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)